### PR TITLE
[Feature]Support reorder ids to split prefill and decodes

### PR DESCRIPTION
**Pull request by YuanRisheng**
## Motivation
为了能够更好的接入三方Attention，需要对输入进行重排，将prefill token和decode token区分开来，本PR支持了重排功能，目前支持了基础场景及投机解码场景下的重排

## Modifications
新增InputBatch结构，用于管理gpu_model_runner的输入，并且添加reorder_split_prefill_and_decode和condense函数支持重排

## Usage or Command
在AttentionMetadata中，通过添加`enable_ids_reorder`字段并设置为True，即可使用P/D重排功能

## Checklist

- [x] Add at least a tag in the PR title.
  - Tag list: [`[FDConfig]`,`[APIServer]`,`[Engine]`, `[Scheduler]`, `[PD Disaggregation]`, `[Executor]`, `[Graph Optimization]`, `[Speculative Decoding]`, `[RL]`, `[Models]`, `[Quantization]`, `[Loader]`, `[OP]`, `[KVCache]`, `[DataProcessor]`, `[BugFix]`, `[Docs]`, `[CI]`, `[Optimization]`, `[Feature]`, `[Benchmark]`, `[Others]`, `[XPU]`, `[HPU]`, `[GCU]`, `[DCU]`, `[Iluvatar]`, `[Metax]`]
  - You can add new tags based on the PR content, but the semantics must be clear.
- [x] Format your code, run `pre-commit` before commit.
- [x] Add unit tests. Please write the reason in this PR if no unit tests.
- [x] Provide accuracy results.
- [ ] If the current PR is submitting to the `release` branch, make sure the PR has been submitted to the `develop` branch, then cherry-pick it to the `release` branch with the `[Cherry-Pick]` PR tag.

### CODE REVIEW

File: custom_ops/gpu_ops/reorder_split_prefill_and_decode.cu [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+    int seq_start = cu_seqlens[i];
+    int prompt_len = prompt_lens[i];
+
+    if (remaining < prompt_len) {
+      return seq_start + remaining;
+    }
+    remaining -= prompt_len;
+  }
+  return -1;  // Should not reach here
+}
```
[the remark this answers is not in the document]
Comment (YuanRisheng) [reply custom_ops/gpu_ops/reorder_split_prefill_and_decode.cu]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/PaddlePaddle/FastDeploy/pull/5194#discussion_r2555286395)

File: custom_ops/gpu_ops/reorder_split_prefill_and_decode.cu [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+        x_ptr,
+        x_reorder_ptr,
+        batch_id_ptr,
+        batch_id_reorder_ptr,
+        cu_seqlens_ptr,
+        prompt_lens_ptr,
+        batch_size,
+        total_decode,
+        total_prefill);
+  }
```
[the remark this answers is not in the document]
Comment (YuanRisheng) [reply custom_ops/gpu_ops/reorder_split_prefill_and_decode.cu]: done

File: custom_ops/gpu_ops/reorder_split_prefill_and_decode.cu (hunk 1 above)
[the remark this answers is not in the document]
Comment (YuanRisheng) [reply custom_ops/gpu_ops/reorder_split_prefill_and_decode.cu]: 这里提出的方案无法实现加速，在第一步，预先在device上计算每个序列的decode/prefill token的累积和的时候，由于累计和前后序列计算的依赖关系，无法并行，依然需要循环batch_size的次数

File: fastdeploy/worker/gpu_model_runner.py [hunk 3]
```diff
@@ -1282,6 +1287,19 @@ def _prepare_inputs(self, is_dummy_or_profile_run=False) -> None:
         # Initialize forward meta data
         self.initialize_forward_meta(is_dummy_or_profile_run=is_dummy_or_profile_run)
 
+        # Reorder inputs to split prefill and decode tokens
+        if self.attn_backends and getattr(self.attn_backends[0].get_attention_meta(), "enable_ids_reorder", False):
+            ids_remove_padding, batch_id_per_token, num_decode = reorder_split_prefill_and_decode(
+                ids_remove_padding, batch_id_per_token, cu_seqlens_q, self.share_inputs["seq_lens_encoder"]
+            )
+            self.share_inputs["num_decode"].copy_(num_decode, False)
```
Comment (gongshaotian) [inline fastdeploy/worker/gpu_model_runner.py]: self.share_inputs["num_decode"] 看起来再其他地方没有用到？

**Review by gongshaotian — DISMISSED** [review]

File: fastdeploy/worker/gpu_model_runner.py (hunk 3 above)
Comment (YuanRisheng) [reply fastdeploy/worker/gpu_model_runner.py]: 暂时还没用到，后续接入其他attention会用

File: fastdeploy/spec_decode/mtp.py [hunk 4]
```diff
@@ -467,65 +457,42 @@ def _init_model_inputs(self):
...
+
+        self.model_inputs["temp_scaled_logprobs"] = paddle.clone(self.target_model_inputs["temp_scaled_logprobs"])
+        self.model_inputs["top_p_normalized_logprobs"] = paddle.clone(
+            self.target_model_inputs["top_p_normalized_logprobs"]
         )
-        self.model_inputs["batch_token_num"] = paddle.full(shape=[self.max_num_seqs], fill_value=0, dtype="int32")
-        self.model_inputs["next_token_num"] = paddle.full(shape=[self.max_num_seqs], fill_value=0, dtype="int32")
+        self.model_inputs["accept_num"] = paddle.clone(self.target_model_inputs["accept_num"])
+        self.model_inputs["accept_tokens"] = paddle.clone(self.target_model_inputs["accept_tokens"])
+        self.model_inputs["draft_logits"] = paddle.clone(self.target_model_inputs["draft_logits"])
```
Comment (zoooo0820) [inline fastdeploy/spec_decode/mtp.py:471]: 这里都改用paddle.clone后，显存是否会相比此前有上涨
